Caltech-IPAC/firefly
# Firefly-677: service descriptors support

### Firefly-677: service descriptors support

  - Better error handling
  - input area for service descriptors that take input
  - Obs core with service descriptors supported
  - Service descriptors in table results supported
  - Non-datalink Obs core that only have one product will be combine with all defined service descriptors
  - handle datalink service descriptors
  - Bug fix: fix some bugs with the new local byte data
  - Feature: Mouse wheel zoom, will now hold the pixel
  - add support for sending timeseries to the timeseries viewer
  - clean up of LcViewer.jsx, convert to functional components


_ticket:_ https://jira.ipac.caltech.edu/browse/FIREFLY-677

### Testing

https://fireflydev.ipac.caltech.edu/firefly-677-servdesc/firefly/?__action=layout.showDropDown&

Using the Tap panel:
- search:  irsa, ztf, ztf_objects_dr3
   - This data set contains one service descriptors
- search: CADC, ivoa, obscore
   - most data products contain some service descriptors (under the `more...` dropdown)
   - most times these down work,  I think it is  a CADC issue, but it is difficult to prove
- upload the file, attached to the ticket:  `ztf-search-result-enhanced-for-ticket.xml`
   - this file has two service descriptors, so you will see the `more...` dropdown
   - try the second one: `show ztf lightcurve in firefly`
